src/uu/cp/src/cp.rs

```diff
@@ -1762,6 +1775,19 @@ fn copy_file(
         permissions
     };
 
+    let source_size = source_metadata.len();
+    let mut use_copy_with_intermediate_progress = false;
+    if source_size >= get_size_limit_for_normal_copy()
+        && !source_is_symlink
+        && !source_is_fifo
+        && symlinked_files.is_empty()
+        && matches!(options.copy_mode, CopyMode::Copy)
+        && progress_bar.is_some()
```

Comment:
Why not check this in `copy_helper`, when most of these conditions have already been checked? This way, the logic to handle each path has to be written only once, and we can't get any weird bugs by having two different paths for the logic to take.

Please move this logic much, *much* closer to whatever eventually calls `fs::copy`.

Also, your copy routine doesn't take advantage of all the OS-level special syscalls and tricks that `fs::copy` might use, so having this diagnostic is going to cost a lot of speed precisely when it matter most: With large files.

Also, keep in mind that some systems/devices can read/write at speeds of gigabytes per second, while others are overwhelmed by a single megabyte per second. Therefore, the threshold is going to be too high and too low at the same time.

Tying this to some kind of time-tracking might be better, especially because that might work independently of whatever `fs::copy` is doing under the hood.